### PR TITLE
fix: validate integer values in `Deno.exitCode` setter

### PR DESCRIPTION
**Pull request by petamoriken**
related: #23609

This PR includes the following:

* Delete the non-executed `cli/tests/unit/os_test.ts` and move its tests to `tests/unit/os_test.ts`
* Fix to validate integer values in `Deno.exitCode` setter (no implicit type conversions)
* Fix to match `process.exitCode` to Node.js behavior (my env: v22.2.0, M1 Mac)

### CODE REVIEW

File: runtime/js/30_os.js [hunk 1]
```diff
@@ -102,10 +102,10 @@ function getExitCode() {
 }
 
 function setExitCode(value) {
-  const code = NumberParseInt(value, 10);
-  if (typeof code !== "number") {
```
Comment (petamoriken) [inline runtime/js/30_os.js:106]: Since the `code` variable is always a number, this validation was dead code.

**Comment by petamoriken** [comment]
@dsherret @bartlomieju please take a look!

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for improving this API!

**Comment by bartlomieju** [comment]
I guess we're blocked by WPT update?

**Review by dsherret — APPROVED** [review]
Thanks!

**Pull request merged**